### PR TITLE
[audaspace] Add new port.

### PR DESCRIPTION
**Pull request by Hoikas**
Adds new port audaspace. All features have been tested on Windows and Linux.

- #### What does your PR fix?  
   See above

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official dynamic triplets

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@Hoikas ,I tested the features in the x64-windows,x64-linux triplet all failed:
````
-- Configuring incomplete, errors occurred!
See also "/home/test/vcpkg/buildtrees/audaspace/x64-linux-dbg/CMakeFiles/CMakeOutput.log".
test@testmachine:~/Jon/test/vcpkg$ cat /home/test/vcpkg/buildtrees/audaspace/config-x64-linux-dbg-err.log
CMake Error at /home/test/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find LibPulse (missing: LIBPULSE_LIBRARY LIBPULSE_INCLUDE_DIR)
Call Stack (most recent call first):
  /home/test/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindLibPulse.cmake:27 (find_package_handle_standard_args)
  /home/test/vcpkg/scripts/buildsystems/vcpkg.cmake:782 (_find_package)
  CMakeLists.txt:618 (find_package)

````
Could you please take a look?

**Comment by Hoikas** [comment]
@JonLiu1993 The `pulseaudio` feature requires the system `libpulse-dev` package to be installed. This matches the behavior of `libsoundio[pulseaudio]`. Please execute `apt-get install libpulse-dev` and try again.

**Comment by JonLiu1993** [comment]
@Hoikas, I installed the libpulse-dev package and tried again, but it still looks like there are errors:
````
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: /home/test/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/test/vcpkg/buildtrees/audaspace/x64-linux-dbg
    See logs for more information:
      /home/test/vcpkg/buildtrees/audaspace/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:102 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/audaspace/portfile.cmake:48 (vcpkg_cmake_install)
  scripts/ports.cmake:141 (include)


Error: Building package audaspace:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: audaspace:x64-linux
  Vcpkg version: 2021-07-26-unknownhash

Additionally, attach any relevant sections from the log files above.
testmachine@17001:~/test/vcpkg$ sudo apt-get install libpulse-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
libpulse-dev is already the newest version (1:8.0-0ubuntu3.15).
0 upgraded, 0 newly installed, 0 to remove and 69 not upgraded.
````
/home/test/vcpkg/buildtrees/audaspace/install-x64-linux-dbg-out.log:
````
/usr/bin/ld: /home/test/vcpkg/installed/x64-linux/debug/lib/libavcodec.a(vc1dsp_mmx.o): relocation R_X86_64_PC32 against symbol `ff_pw_9' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status

````

**Comment by Hoikas** [comment]
Static triplets (x64-linux) are not supported.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/audaspace/0001-Remove-C-CXX-flags.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ 
+ if(MSVC)
+-	list(APPEND CMAKE_C_FLAGS_DEBUG "/Zi /Od")
+-	list(APPEND CMAKE_CXX_FLAGS_DEBUG "/Zi /Od")
+-	list(APPEND CMAKE_SHARED_LINKER_FLAGS_DEBUG "/DEBUG")
+-	list(APPEND CMAKE_STATIC_LINKER_FLAGS_DEBUG "/DEBUG")
+-	list(APPEND CMAKE_EXE_LINKER_FLAGS_DEBUG "/DEBUG")
+ 	add_definitions(
+ 		/D_USE_MATH_DEFINES
+ 		/EHsc
```
Comment (ras0219) [inline ports/audaspace/0001-Remove-C-CXX-flags.patch:30]: Fwiw we might want to remove `/EHsc` as well

**Comment by ras0219** [comment]
If we aren't building `FFMPEG` with `-fPIC`, that's actually a bug. We intend our default static triplets to always build with `-fPIC`.

**Comment by Hoikas** [comment]
> If we aren't building FFMPEG with -fPIC, that's actually a bug. We intend our default static triplets to always build with -fPIC.

I was wondering about that but figured there was a performance reason for not defaulting to -fPIC. I've seen this in another port that consumes ffpmeg on linux, but I don't remember which one offhand.

**Comment by cenit** [comment]
> > If we aren't building FFMPEG with -fPIC, that's actually a bug. We intend our default static triplets to always build with -fPIC.
> 
> I was wondering about that but figured there was a performance reason for not defaulting to -fPIC. I've seen this in another port that consumes ffpmeg on linux, but I don't remember which one offhand.

#17298 was adding - fPIC exactly for the same error message. Is the flag not working anymore in recent ffmpeg setup scripts?

**Comment by JonLiu1993** [comment]
@Hoikas ，Is this pr now ready for review?

**Comment by Hoikas** [comment]
Yes, it has been ready since July 31.

**Comment by JonLiu1993** [comment]
> Yes, it has been ready since July 31.

Sorry for the delaly

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@Hoikas ,When I tested the feature locally , it's failed. Could you please take a look?
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/7142478/config-x86-windows-out.log)

**Comment by cenit** [comment]
@Hoikas usually this line is mandatory in a downstream project consuming ffmpeg (static) that builds a shared library.
Do you use it?

```
set(CMAKE_SHARED_LINKER_FLAGS "-Wl,-Bsymbolic")
```

**Pull request closed without merging**